test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2101,9 +2101,24 @@ protected static SecureRandom secureRandomFips(final byte[] seed) throws NoSuchA
         return secureRandomFips;
     }
 
+    /**
+     * The timeout used for the various "safe" wait methods such as {@link #safeAwait} and {@link #safeAcquire}. In tests we generally want
+     * these things to complete almost immediately, but sometimes the CI runner executes things rather slowly so we use {@code 10s} as a
+     * fairly relaxed definition of "immediately".
+     * <p>
+     * A well-designed test should not need to wait for longer than this. If you think you need to do so, instead seek a better way to write
+     * the test such that it does not need to wait for so long. Tests that take 10s of seconds to complete are a big drag on CI times which
+     * slows everyone down.
```

Comment:
I wonder if we could make it even more restrictive?
I would say a unit test should not last longer than a second even on possibly slow CI.
For integration tests 10s sounds more reasonable.

Reply:
Even integ tests generally should run faster than this IMO. I strengthened the wording a little in ca73c5bde2a.